### PR TITLE
Vault upgrade

### PR DESCRIPTION
**Pull request by PetersonBS**

### CODE REVIEW

File: app/vault_client.py [hunk 1]
```diff
@@ -63,49 +67,55 @@ def post(self):
...
+        except Exception as error:
+            app.logger.exception(error)
+            resp = json_response.create(HttpCode.SERVER_ERROR.value, HttpBody.INIT_VAULT_FAIL.value)
             return resp
 
+        root_token = vault['root_token']
+        unseal_keys = vault['keys']
+
+        try:
+            _write_token(root_token)
```
Comment (kroozo) [inline app/vault_client.py]: Why is this module level?

File: app/vault_client.py [hunk 2]
```diff
@@ -61,55 +65,57 @@ def post(self):
...
 
-        shares = marshal_json['shares']
-        threshold = marshal_json['threshold']
+        shares = vault_params['shares']
+        threshold = vault_params['threshold']
 
         if threshold > shares or shares >= 2 and threshold == 1 or shares < 1 or threshold < 1:
-            resp = json_response.create(HttpCode.bad_request.value, HttpBody.init_vault_fail_due_to_parameter.value)
+            resp = json_response.create(HttpCode.BAD_REQUEST.value, HttpBody.INIT_VAULT_BAD_REQUEST.value)
             return resp
```
Comment (kroozo) [inline app/vault_client.py]: I don't think we ever need resp. It gets returned immediately everywhere
```suggestion
            return json_response.create(HttpCode.BAD_REQUEST.value, HttpBody.INIT_VAULT_BAD_REQUEST.value)
```

File: app/vault_client.py [hunk 3]
```diff
@@ -61,55 +65,57 @@ def post(self):
...
-                        unseal_keys_file.write("%s\n" % key)
-            except IOError as error:
-                app.logger.exception(error)
-                resp = json_response.create(HttpCode.server_error.value, HttpBody.init_vault_fail.value)
-                return resp
-
-            resp = json_response.create(HttpCode.created.value, HttpBody.init_vault_success.value)
+
+        try:
+            _write_unseal_keys(unseal_keys)
```
Comment (kroozo) [inline app/vault_client.py]: Why is this module level?

File: app/vault_client.py [hunk 4]
```diff
@@ -154,87 +169,134 @@ def get(self, secret_name):
...
+            return resp
+
+        try:
+            secret = client.secrets.kv.v1.read_secret(path=secret_name)
+        except hvac.exceptions.InvalidPath as error:
+            _seal_vault(client)
+
+            resp = json_response.create(HttpCode.NOT_FOUND.value, HttpBody.SECRET_NOT_EXIST.value)
+            return resp
+        except Exception as error:
```
Comment (kroozo) [inline app/vault_client.py:207]: I think this should call `_seal_vault` as well. Probably with `finally` as elsewhere.

**Review by kroozo — CHANGES_REQUESTED** [review]

File: app/vault_client.py (hunk 4 above)
Comment (PetersonBS) [reply app/vault_client.py:207]: We only need to seal the vault, if the key was not found. If the key is there, we continue with an update. That is why it is not in the finally block, but just before the return.

File: app/vault_client.py (hunk 2 above)
Comment (PetersonBS) [reply app/vault_client.py]: Agreed.

File: app/vault_client.py (hunk 1 above)
Comment (PetersonBS) [reply app/vault_client.py]: Will factor out initialization thus there will be no need for package-visible connection data.

File: app/vault_client.py (hunk 3 above)
Comment (PetersonBS) [reply app/vault_client.py]: Will factor out initialization thus there will be no need for package-visible connection data.

**Pull request merged**